### DISCUSSION TITLE
compile DB file lookup

### DISCUSSION THREAD

**Discussion by mellery451** [clangd/clangd#1301] [Q&A] [answered]
I've just upgraded to clangd 15 (homebrew linux) and I think file lookups in my compile DB are now failing. The compile DB has file paths relative to the workspace root - e.g:

    "file": "some/path/in/my/repo/source.cpp"

and it looks like my lsp client (nvim + lspconfig) are sending requests with:

    "uri":"file:///PATH/TO/REPO/some/path/in/my/repo/source.cpp",

There is no explicit indication that the lookup fails, but I see that clangd compiles with default args. If I modify an entry in the DB to have the fullpath (that matches the uri above), then it finds the entry and it gets the right flags.

I swear this used to work...maybe my client has changed or is it possible that clangd has changed the lookup logic such that it doesn't handle relative entries in the DB any more? 

I should also mention that working directory seems to be what I would expect such that clangd should be able to compare full and relative paths - from the clangd startup message:

    Working directory: /PATH/TO/REPO

**Comment by i-ky** [comment, marked as answer]
What is `"directory"` in your compilation database? According to [JSON Compilation Database Format Specification](https://clang.llvm.org/docs/JSONCompilationDatabase.html):
> **directory**: The working directory of the compilation. All paths specified in the **command** or **file** fields must be either absolute or relative to this directory.

**Reply by mellery451** [reply]
this was it...thanks for reminding me. I forgot that I have to run a post-processing cleanup step after generating my DB to properly set some of these fields. That resolved it.
